### PR TITLE
[Service Fabric] Add managed cluster and node type commands

### PR DESCRIPTION
**Pull request by a-santamaria**
**Description**  

These cmdlets will support a New offering (resource type) - Service Fabric managed clusters for our customers. In this new model, Service Fabric resource provider (SFRP) manages the entire resource group for the essential components of the cluster, including VMSS, VNet/Load Balancer, and also storage account for diagnostics and bootstrap. In the previous model, SFRP manages a "cluster" resource that is mostly about the configurations about the cluster, the actual cluster runs on VMSS / VNet / etc that the customer needs to deploy in in parallel with the cluster resource and those are directly managed by the customer on their own.

**Testing Guide**  
see src\azure-cli\azure\cli\command_modules\servicefabric\_help.py for examples and descriptions

**History Notes**  

**ServiceFabric**
* Adding commands for managed clusters and node types.
    - sf managed-cluster
    - sf managed-cluster client-certificate
    - sf managed-node-type
    - sf managed-node-type node
    - sf managed-node-type vm-extension
    - sf managed-node-type vm-secret
* Upgraded Service Fabric SDK to version 0.5.0 which uses service fabric resource provider api-version 2020-03-01 for the current model and 2020-01-01-preview for managed clusters.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Fabric

**Comment by yungezz** [comment]
hi @qwordy could you pls help to review? seems the PR need to catch Ignite timeline

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
+**ServiceFabric**
+
+* Adding commands for managed clusters and node types.
+    - sf managed-cluster
+    - sf managed-cluster client-certificate
+    - sf managed-node-type
+    - sf managed-node-type node
+    - sf managed-node-type vm-extension
+    - sf managed-node-type vm-secret
+* Upgraded Service Fabric SDK to version 0.5.0 which uses service fabric resource provider api-version 2020-03-01 for the current model and 2020-01-01-preview for managed clusters.
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:15]: Not necessary any more. History notes will be generated from PR title or description.

File: src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_node_types.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                     vm_image_version=None,
+                     capacity=None,
+                     placement_property=None):
+
+    #  set defult parameters
+    if disk_size is None:
+        disk_size = 100
+
+    if vm_size is None:
+        vm_size = "Standard_D2"
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_node_types.py:47]: Pay attention to case. Maybe case insensitive is better.

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/HISTORY.rst (hunk 1 above)
Comment (a-santamaria) [reply src/azure-cli/HISTORY.rst]: ok thanks I have removed it

File: src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_node_types.py (hunk 2 above)
Comment (a-santamaria) [reply src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_node_types.py:47]: doing it the same way they have it in the examples of the sdk documentation. it's working correctly

**Pull request merged**